README.md

```diff
@@ -1,51 +1,199 @@
+        let handle = payload.dictionaryPayload["handle"] as? String ?? ""
+        let isVideo = payload.dictionaryPayload["isVideo"] as? Bool ?? false
+        
+        let data = flutter_callkeep.Data(id: id, callerName: callerName, handle: handle, hasVideo: isVideo)
+        //set more data
+        data.extra = ["user": "abc@123", "platform": "ios"]
+        //data.iconName = ...
+        //data.appName = ...
+        //data.....
+        SwiftCallKeepPlugin.sharedInstance?.displayIncomingCall(data, fromPushKit: true)
+    }   
+}
 ```
-```xml
-<application>
-    ..
-    <service android:name="io.wazo.callkeep.VoiceConnectionService"
-        android:label="Wazo"
-        android:permission="android.permission.BIND_TELECOM_CONNECTION_SERVICE">
-        <intent-filter>
-            <action android:name="android.telecom.ConnectionService" />
-        </intent-filter>
-    </service>
-    ..
-</application>
+
+## Usage
+
+### Setup: 
+
+You need to have base `CallKeep` config setup to reduce code duplication and make it easier to display incoming calls: 
+
+```dart
+ final callKeepBaseConfig = CallKeepBaseConfig(
+      appName: 'Done',
+      androidConfig: CallKeepAndroidConfig(
+        logo: 'logo',
+        notificationIcon: 'notification_icon',
+        ringtoneFileName: 'ringtone.mp3',
+        accentColor: '#34C7C2',
+      ),
+      iosConfig: CallKeepIosConfig(
+        iconName: 'Icon',
+        maximumCallGroups: 1,
+      ),
+    );
+```
+
+### Display incoming call:
+
+```dart
+// Config and uuid are the only required parameters
+final config = CallKeepIncomingConfig.fromBaseConfig(
+    config: callKeepBaseConfig,
+    uuid: uuid,
+    contentTitle: 'Incoming call from Done',
+    hasVideo: hasVideo,
+    handle: handle,
+    callerName: incomingCallUsername,
+    extra: callData,
+);
+await CallKeep.instance.displayIncomingCall(config);
+```
+
+### Show missed call notification (Android only):
+
+```dart
+// config and uuid are the only required parameters
+final config = CallKeepIncomingConfig.fromBaseConfig(
+    config: callKeepBaseConfig,
+    uuid: uuid,
+    contentTitle: 'Incoming call from Done',
+    hasVideo: hasVideo,
+    handle: handle,
+    callerName: incomingCallUsername,
+    extra: callData,
+);
+await CallKeep.instance.showMissCallNotification(config);
+```
+
+### Start an outgoing call:
+
+```dart
+// config and uuid are the only required parameters
+final config = CallKeepOutgoingConfig.fromBaseConfig(
+    config: DoneCallsConfig.instance.callKeepBaseConfig,
+    uuid: uuid,
+    handle: handle,
+    hasVideo: hasVideo ?? false,
+);
+CallKeep.instance.startCall(config);
 ```
 
-if you want to use the function `displayCustomIncomingCall`, please add the following permission to the `AndroidManifest.xml`.
+### Handling events:
 
-```xml
-<uses-permission android:name="android.permission.USE_FULL_SCREEN_INTENT" />
+```dart
+CallKeep.instance.onEvent.listen((event) async {
+    // TODO: Implement other events
```

Comment:
Which event

Reply:
You can see it on the docstring of the `onEvent` or if you remove the `default` switch statement.